source/agora/consensus/pool/Transaction.d

```diff
@@ -570,6 +580,29 @@ public class TransactionPool
     {
         log.error("SQLite error: ({}) {}", code, msg.fromStringz);
     }
+
+    /***************************************************************************
+
+        Get an UTXO, no double-spend protection.
+
+        Params:
+            hash = the hash of the UTXO (`hashMulti(tx_hash, index)`)
+            value = the UTXO
+
+        Returns:
+            true if the UTXO was found
+
+    ***************************************************************************/
+
+    public bool peekUTXO (in Hash utxo, out UTXO value) nothrow @safe
+    {
+        if (auto output = utxo in this.utxo_set)
+        {
+            value.output = *output;
+            return true;
+        }
+        return false;
+    }
```

Comment:
Looks to me like we should just have another `UTXOCache` here ?

Reply:
That was my initial thought too, but to remove a UTXO we would either need to construct a dummy spending TX or expose the private `remove` UTXOCache has. Either way didnt feel right, so this.

I can change it to use UTXOCache if you are comfortable with one of those.

Reply:
That answers my question